t/basic.t

```diff
@@ -638,4 +638,160 @@ subtest "duplicated creation ids" => sub {
   );
 };
 
+subtest "datetime field validations" => sub {
+  my $tsrez = re(qr/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/);
+
+  my $res = $jmap_tester->request([
+    [
+      setCookies => {
+        create    => {
+          yellow => { type => 'yellow' }, # default baked_at
+          gold   => { type => 'gold', baked_at => undef }, # null baked_at
+          blue   => { type => 'blue', baked_at => '2016-01-01T12:34:56Z' },
+          white  => { type => 'white', baked_at => '2016-01-01T12:34:57Z' },
+          red    => { type => 'red', baked_at => '2016-01-01T12:34:56' },
+          green  => { type => 'green', expires_at => undef },
+          pink   => { type => 'pink', expires_at => '2016-01-01T12:34:56Z' },
+          onyx   => { type => 'onyx', expires_at => '2016-01-01T12:34:57Z' },
+        },
+      },
+    ],
+  ]);
+
+  cmp_deeply(
+    $jmap_tester->strip_json_types( $res->as_pairs ),
+    [
+      [
+        cookiesSet => superhashof({
+          oldState => 16,
+          newState => 17,
+          created => {
+            yellow => { id => ignore(), expires_at => ignore(), baked_at => $tsrez },
+            gold   => { id => ignore(), expires_at => ignore(), },
+            blue   => { id => ignore(), expires_at => ignore(), },
+            white  => { id => ignore(), expires_at => ignore(), },
+            pink   => { id => ignore(), baked_at => ignore(), },
+            onyx   => { id => ignore(), baked_at => ignore(), },
+          },
+          notCreated => {
+            red => superhashof({
+              type => 'invalidProperties',
+              propertyErrors => { baked_at => 'invalid date value' },
+            }),
+            green => superhashof({
+              type => 'invalidProperties',
+              propertyErrors => { expires_at => 'no value given for required field' },
+            }),
```

Comment:
I am going to be a **monster** and move the goal posts here.

In fact, you did give a value: you gave `null`.  If the field had been missing, it would've gotten a default.  Do you think we can/should give a better error message here?  I bet we can and should.  Something like "null value given but [type] required".  Legit?